Added separate linking from primary image panel

joeletherton opened the pull request:
Added image linking directly for the panel. Added 2 options to support this linking (described below).
The panel itself becomes a clickable link which can redirect to any url specified in the data-href attribute
of the original image html.

New options:
    - panel_clickable
        - BOOLEAN: flag to indicate that the panel should have a clickable link
    - link_newwindow
        - BOOLEAN: flag to indicate that panel links should/should not open in a new window

Usage-script:

```
    $(function () {
        $('#myGallery').galleryView({
            panel_clickable: true,
            link_newwindow: false
        });
    });
```

Usage-html:

```
    <li><img src="myImage.jpg" data-href="http://www.google.com/" />
```

joeletherton committed:
Update master

joeletherton committed:
Update master

joeletherton committed:
Update to add clickable links. Added updateClickable method, new options, updateClickable binding, and this.href setting.

joeletherton committed:
Added actual call for updateClickable to fire on the change of image.

joeletherton committed:
Added "undelegate" to panelWrap. Stacking delegates caused multiple click events to fire simultaneously resulting in a lot of open windows.

joeletherton committed:
Added randomness and applicable option to filmstrip showNext when "playing".

joeletherton committed:
Added changelog entry for random setting

greg01 commented:
Is it possible that the mouse pointer over the driving of the picture changes to a finger. Otherwise, the user has no idea that this is a picture link.

But one can also think about css:
. gv_panel img {position: absolute; cursor: pointer;}

joeletherton committed:
Changed option "panel_clickable" boolean into "clickable" string to identify which item to link. Used "all" as an option to enable expansion of this linking property in the future.

joeletherton committed:
Added new cursor_pointer style to allow changing the cursor on a single element.

joeletherton committed:
Removed option "panel_clickable" and added option "clickable"

Didi23 commented:
Thanks for great addition to primary panel images, but is it possible to make separate links from filmstrip? Maybe add filmstrip to "clickable" string?

joeletherton commented:
@Didi23: I'm not sure what you mean. Are you referring to making the filmstrip images clickable to the alternate link?

Didi23 commented:
@joeletherton  Yes. I have 2 different images, in primary panel I have advert and in filmstrip I have a brand name thumbnail image so people tend to click on filmstrip instead of primary panel to get to the page with that brand. So I need filmstrip images to be clickable to the alternate link. Can you please help me do it?

joeletherton committed:
Update js/jquery.galleryview-3.0-dev.js

Added an option "filmstrip_use_alt_click" to enable the film strip
to be clickable in the same fashion as other images.

joeletherton commented:
@Didi23: I added a new option to the code and some logic to operate it. If you set the boolean "filmstrip_use_alt_click" to true and the image clicked has a non-empty string in the data-href attribute, then it'll behave as if the primary image was clicked. Otherwise it will use default behavior.

Didi23 commented:
@joeletherton  Thank you so much!!! It is exactly what I needed. You made my day. Again, thanks a lot :)
